cryptoki/src/functions/slot_token_management.rs

```diff
@@ -47,6 +45,47 @@ impl Pkcs11 {
         Ok(slots)
     }
 
+    /// Get all slots available with a token
+    pub fn get_slots_with_initialized_token(&self) -> Result<Vec<Slot>> {
+        let mut slot_count = 0;
+
+        unsafe {
+            Rv::from(get_pkcs11!(self, C_GetSlotList)(
+                cryptoki_sys::CK_TRUE,
+                std::ptr::null_mut(),
+                &mut slot_count,
+            ))
+            .into_result()?;
+        }
+
+        let mut slots = vec![0; slot_count.try_into()?];
+
+        unsafe {
+            Rv::from(get_pkcs11!(self, C_GetSlotList)(
+                cryptoki_sys::CK_TRUE,
+                slots.as_mut_ptr(),
+                &mut slot_count,
+            ))
+            .into_result()?;
+        }
+
+        let mut slots: Vec<Slot> = slots.into_iter().map(Slot::new).collect();
+
+        // This should always truncate slots.
+        slots.resize(slot_count.try_into()?, Slot::new(0));
```

Comment:
I think you can replace all those lines with:
```rust
let slots = self.get_slots_with_token()?;
```
It should do the same using the method above!